Feat: hide pool record if base balance of a pool is too small

shallcdn opened the pull request:
- [x] Add logic to check if the  pool share value less than 0.01 percent, the pool record will be hidden

Ref: #884

lijac commented on src/app/components/trade/LiquidityDetails.tsx:
```diff
@@ -219,19 +220,25 @@ const PoolRecord = ({
-      </DataText>
-    </ListItem>
+    <>
+      {parseFloat(share.multiply(100).toFixed(4)) > LOWEST_LIQUIDITY_POOL_PERCENT ? (
```
Can we use comparison methods of `Fraction` directly? @shallcdn

lijac replied:
`share.multiply(100).greaterThan(LOWEST_LIQUIDITY_POOL_PERCENT_FRAC)`
and
`LOWEST_LIQUIDITY_POOL_PERCENT_FRAC = new Fraction(1, 100)`

shallcdn replied:
@lijac  Yes, we can, but I wonder why don't we use pure JS as the way I wrote the codes? Do we have any reasons? 
Ext: easy to read, consistency... I just want to know the reason to prevent my mistakes for any PRs in the future.

lijac replied:
Sure, here are the reasons I can think of:
1. We may need to update `Fraction` class in the future and that can be `toFixed` method. If that's the case, we will probably update `greaterThan` to accommodate new changes but cannot guarantee `toFixed(4)` still returns the desired value.
2. We also have `BigNumber` in the project for processing numbers (though it is going to fade out), it's better to use explicit functions of separate classes to prevent confusion.

Personal preference also has a place in my mind :)

shallcdn replied:
@lijac I have updated, Please help take a look again, thank you!

lijac commented:
@BennyOptions I assume we don't hide ICX/sICX pool even if pool share % is less than 0.01%. Can you confirm it?

BennyOptions commented:
@shallcdn my apologies, I should have shared more detailed logic. Using the pool-share percentage is dangerous because average users participating in a large pool will likely have small percentages of the total pool and their contribution will be hidden.

Please use a different logic to hide small values. I would prefer this logic:

If (quoteDepositAmount < 0.01 && baseDepositAmount < 0.01)
  then hidePool

octocat512 commented:
Instead of hiding the small amount, I'd like to display the portions as lesser than 0.01%.
for example: <0.01%

BennyOptions commented:
https://github.com/balancednetwork/balanced-network-interface/pull/895#issuecomment-1054596981

We will use the above linked solution for now, and base it on **right-side currency (quote currency)**. So if quote currency amount is less than 0.01, hide it.

shallcdn commented:
> [#895 (comment)](https://github.com/balancednetwork/balanced-network-interface/pull/895#issuecomment-1054596981)
> 
> We will use the above linked solution for now, and base it on **right-side currency (quote currency)**. So if quote currency amount is less than 0.01, hide it.

Hi @BennyOptions let me confirm again your comment:
![Screen Shot 2022-03-02 at 17 35 24](https://user-images.githubusercontent.com/99955678/156345422-deb03e3e-3619-48ab-86a5-d67342517ee2.png)

With the logic 
```
If (quoteDepositAmount < 0.01 && baseDepositAmount < 0.01)
then hidePool
```
I understand your mean and based on the image above:

_quoteDepositAmount_ stand for sICX value
_baseDepositAmount_ stand for bnUSD value

Isn't it?

>  and base it on right-side currency (quote currency). So if quote currency amount is less than 0.01, hide it.

so which column's value do you want to mention?

BennyOptions commented:
@shallcdn my apologies, I always get base and quote confused. I will speak only in terms of left-side currency and right-side currency. As an example for sICX / bnUSD, sICX = left-side currency and bnUSD = right-side currency.

The logic I would like should be based only on right-side currency. The reason is that there are constraints at the smart contract level around which currencies are allowed to be right-side currencies. This means we can be comfortable in choosing nominal constraints for right-side currency, such as < 0.01. The specific logic I recommend is below, with the understanding that baseDepositAmount represents the right-side currency:

```
If (baseDepositAmount < 0.01)
then hidePool
```

shallcdn commented:
@lijac li I have updated the logic of this MR based on the review of @BennyOptions. Please help retake a look at one.

one more thing is that the function greaterThan doesn't work as we expected, so I changed to use it as usual way comparison

shallcdn commented:
@lijac l have updated logic based on the [comment](https://iconcommunity.slack.com/archives/G0117R69JK0/p1646424448973689?thread_ts=1645848889.366999&cid=G0117R69JK0 ) and squash commit
Please help take a look. Thank you!

lijac reviewed (approved):
LGTM, ready to merge.

BennyOptions commented:
ready to merge - i checked UX

eulerbeat commented:
Close #909
